crawl-ref/source/tags.cc

```diff
@@ -3926,6 +3926,15 @@ static void _tag_read_you_items(reader &th)
             // Items in inventory have already been handled.
             if (th.getMinorVersion() < TAG_MINOR_ISFLAG_HANDLED)
                 it.flags |= ISFLAG_HANDLED;
+
+            // Recurse everything for Ash worshippers, since we
+            // lost old curse information during unmarshall cleanup.
+            if (th.getMinorVersion() < TAG_MINOR_UNCURSE
+                && you_worship(GOD_ASHENZARI)
+                && item_type_has_curses(it.base_type))
+            {
+                do_curse_item(it);
+            }
```

Comment:
This looks like it curses everything in an Ash worshipper's inventory. This seems worse(?) than the alternative (uncurse everything) since Ash followers sometimes intentionally leave things uncursed. I guess re or un-cursing is going to have to happen either way, and either way it'll be a surprise.

Reply:
It only takes one ?id to uncurse everything, but many ?id to recurse everything!